presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryMetadata.java

```diff
@@ -199,6 +201,15 @@ public synchronized MemoryOutputTableHandle beginCreateTable(ConnectorSession se
         return new MemoryOutputTableHandle(table, ImmutableSet.copyOf(tableIds.values()));
     }
 
+    private void assertTableNotExists(SchemaTableName tableName)
+    {
+        if (tables.values().stream()
+                .map(MemoryTableHandle::toSchemaTableName)
+                .anyMatch(tableName::equals)) {
```

Comment:
`AtomicLong` assures that tableIDs are unique, this ensures that table names are unique.